nexus/db-model/src/external_ip.rs

```diff
@@ -78,6 +79,18 @@ impl From<InstanceExternalIp> for sled_agent_client::types::SourceNatConfig {
     }
 }
 
+/// Describes where the IP candidates for allocation come from: either
+/// from an IP pool, or from a project.
+///
+/// This ensures that a source is always specified, and a caller cannot
+/// request an external IP allocation without providing at least one of
+/// these options.
+#[derive(Debug, Clone, Copy)]
+pub enum IpSource {
```

Comment:
Nice, this makes it much easier to see the situations under which each of the pool/project IDs are required or not. Thanks!